### ISSUE TITLE
Train model to detect cord compression

### ISSUE THREAD

**Issue by jcohenadad** [spinalcordtoolbox/spinalcordtoolbox#4333]
The idea is to train a model that runs on axial MRI slices, and that detects if a slice has a compression. 

<img width="637" alt="image" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/0e3cc0d2-b1dc-45e8-88cf-f5ec865cf03d">

Data: `dcm-zurich` (internal)

Biblio:
- https://pubmed.ncbi.nlm.nih.gov/34006910/
- https://pubmed.ncbi.nlm.nih.gov/35371944/ -- implemented in [SCT #4760](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4760)

Related to:
- https://github.com/ivadomed/ivadomed/issues/762
- https://github.com/neuropoly/idea-projects/issues/2
- https://github.com/spinalcordtoolbox/detect-compression
- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4333

**Comment by sandrinebedard** [comment]
we also had what what Étienne last summer was working on? maybe we can cross-link!

**Comment by valosekj** [comment]
Related: https://github.com/neuropoly/idea-projects/issues/2, https://github.com/spinalcordtoolbox/detect-compression

(I am also exploring several approaches by myself, so far in the very very beginning stage -- code is not ready to be pushed)

**Comment by jcohenadad** [comment]
right-- i guess i was thinking more of an approach where we analyse the image data directly (as opposed to training a model from the CSA).

**Comment by naga-karthik** [comment]
well well! glad that I found this issue sooner than later, thanks @valosekj for linking it! 

my ideas were related to dcm lesion segmentation, for which, I was planning on doing _supervised pre-training_ on `dcm-zurich` dataset (because we have more subjects here than in `dcm-zurich-lesions-*` datasets) to predict SC seg and compression sites. Hopefully, this pre-trained model will be useful towards segmenting dcm lesions.
